modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneRebalanceRaftGroupEventsListener.java

```diff
@@ -597,6 +597,9 @@ public static CompletableFuture<Void> handleReduceChanged(
         ByteArray changeTriggerKey = ZoneRebalanceUtil.pendingChangeTriggerKey(partId);
         byte[] rev = ByteUtils.longToBytesKeepingOrder(entry.revision());
 
+        ByteArray changeTimestampKey = ZoneRebalanceUtil.pendingChangeTriggerKey(partId);
```

Comment:
Why did you introduce new key? can't we reuse ` changeTriggerKey`?

Reply:
But this is the one that existed before

Reply:
why do you need *changeTimestampKey* and why do you need further changes like this 
```
        Condition changeRevisionAndTimestampDontExistOrLessThan = and(
                or(notExists(changeTriggerKey), value(changeTriggerKey).lt(rev)),
                or(notExists(changeTimestampKey), value(changeTimestampKey).lt(timestamp))
        );
```